dirty_cat/_fuzzy_join.py

```diff
@@ -31,6 +36,7 @@ def fuzzy_join(
     analyzer: Literal["word", "char", "char_wb"] = "char_wb",
     ngram_range: Tuple[int, int] = (2, 4),
     return_score: bool = False,
+    encoder: Literal["hashing", "count"] = "hashing",
```

Comment:
The type hint would probably be 
```suggestion
    encoder: Union[Literal["hashing", "count"], _VectorizerMixin] = "hashing",
```
That's true that it forces us to import from the private sklearn API. However, it's not a problem in this case: the public API of a library is *what can be imported by the user in a production environment*. The public API ensures the changes to be documented and to some extent, backward-compatible. Everything private is internal API to the lib, thus is not assured to be backward-compatible, nor useful for the standard user. We are developing a library, so it's a different use-case of sklearn.
I hope my explaination makes sense :sweat_smile: